core/src/genesis.rs

```diff
@@ -500,7 +505,7 @@ mod tests {
             let domain_id: DomainId = "wonderland".parse().unwrap();
             assert_eq!(
                 finished_genesis_block.transactions[0].isi[0],
-                RegisterBox::new(domain_id.clone()).into()
+                Instruction::from(RegisterBox::new(Domain::new(domain_id.clone())))
```

Comment:
The CI doesn't know that what we should have is a `NewDomain` and not a `Domain`. We didn't catch this error because it wasn't reflected in the `genesis.json`, because both can be deserialised equally well.